packages/react-query/src/__tests__/useQuery.test.tsx

```diff
@@ -2231,7 +2228,28 @@ describe('useQuery', () => {
     // 7. Observer options updated
     // 8. Observer result updated -> stale
     // 9. Observer options updated
-    expect(fn).toHaveBeenCalledTimes(9)
+
+    await waitFor(() => {
+      //here query succeeds
+      expect(fn).toHaveBeenCalledWith(
+        expect.objectContaining({
+          type: 'updated',
+          action: expect.objectContaining({ type: 'fetch' }),
+        }),
+      )
+    })
+
+    await waitFor(() => {
+      // here query goes stale
+      expect(fn).toHaveBeenCalledWith(
+        expect.objectContaining({
+          type: 'updated',
+          action: expect.objectContaining({ type: 'success' }),
+        }),
+      )
+    })
+
+    unsubscribe()
   })
 
   it('should not re-render when it should only re-render on data changes and the data did not change', async () => {
```

Comment:
this test is also failing now. It wasn't change so I think it's just flaky too:

```
packages/react-query test:jest: FAIL react-query src/__tests__/useQuery.test.tsx (31.7 s)
packages/react-query test:jest:   ● useQuery › should not re-render when it should only re-render only data change and the selected data did not change
packages/react-query test:jest:     expect(received).toBe(expected) // Object.is equality
packages/react-query test:jest:     Expected: 2
packages/react-query test:jest:     Received: 1
packages/react-query test:jest:       1038 |     await sleep(10)
packages/react-query test:jest:       1039 |
packages/react-query test:jest:     > 1040 |     expect(states.length).toBe(2)
packages/react-query test:jest:            |                           ^
packages/react-query test:jest:       1041 |     expect(states[0]).toMatchObject({ data: undefined })
packages/react-query test:jest:       1042 |     expect(states[1]).toMatchObject({ data: 'test' })
packages/react-query test:jest:       1043 |   })
packages/react-query test:jest:       at Object.<anonymous> (src/__tests__/useQuery.test.tsx:1040:27)
```

Reply:
[2630a1f](https://github.com/TanStack/query/pull/4466/commits/2630a1f4d45e3b33df87cd266e7a0de7a6bc5e1c)